src/stores/authStore.ts

```diff
@@ -0,0 +1,78 @@
+import { create } from 'zustand';
+import { createJSONStorage, persist } from 'zustand/middleware';
+import { devtools } from 'zustand/middleware';
+
+const AUTH_STORE_NAME = 'auth-state';
+
+interface AuthState {
+  isLoggedIn: boolean;
+  accessToken: string | null;
+  refreshToken: string | null;
+  login: (accessToken: string, refreshToken: string) => void;
+  logout: () => void;
+  setAccessToken: (newAccessToken: string) => void;
+  setRefreshToken: (newRefreshToken: string) => void;
+}
+
+export const useAuthStore = create<AuthState>()(
+  devtools(
+    persist(
+      (set) => ({
+        isLoggedIn: false,
+        accessToken: null,
+        refreshToken: null,
+        login: (accessToken, refreshToken) =>
+          set({
+            isLoggedIn: true,
+            accessToken,
+            refreshToken,
+          }),
+        logout: () =>
+          set({
+            isLoggedIn: false,
+            accessToken: null,
+            refreshToken: null,
+          }),
+        setAccessToken: (newAccessToken) => set({ accessToken: newAccessToken }),
+        setRefreshToken: (newRefreshToken) => set({ refreshToken: newRefreshToken }),
+      }),
+      {
+        name: AUTH_STORE_NAME,
+        storage: createJSONStorage(() => localStorage),
+      },
+    ),
+  ),
+);
+
+// 페이지 접속 시 최초 실행되며 로컬스토리지에 이전 사용 토큰이 있는지 확인, 있다면 해당 값 사용
+// node 서버에서의 실행 방지를 위해 window 객체가 존재하는 브라우저일 때만 코드 실행
+if (typeof window !== 'undefined') {
+  // 로컬 스토리지에서 직접 상태를 가져오는 함수
+
+  const getAuthStateFromStorage = () => {
+    if (typeof window !== 'undefined') {
+      const storedState = localStorage.getItem(AUTH_STORE_NAME);
+      if (storedState) {
+        try {
+          const parsedState = JSON.parse(storedState);
+          return parsedState.state as AuthState;
+        } catch (error) {
+          console.error('인증 로컬 저장소 parsing 실패: ', error);
+        }
+      }
+    }
+    return null;
+  };
+
+  const initializeAuth = () => {
+    const storedState = getAuthStateFromStorage();
+
+    if (storedState && storedState.accessToken && storedState.refreshToken) {
```

Comment:
혹시 `useAuthStore`만 이렇게 사용하신 이유가 있을까요?!
storedState처럼 추가하고 아래처럼 사용하는게 어떨까요?
그리고 여기 옵셔널 체이닝 쓰는게 더 간결할 것 같습니다!

```suggestion
    if (storedState?.accessToken && storedState?.refreshToken) {
      authStore.login(storedState.accessToken, storedState.refreshToken);
    }
  };
```

Reply:
1. 옵셔널 체이닝으로 수정하겠씁니다!

2. 이 부분은 제가 좀 잘못생각했네요. 토큰 값을 zustand + persist로 관리하니 zustand에서 알아서 다 해주는건데 .. 처음 생각했을 때는 사용자가 페이지에 접속했을 때, 로컬 스토리지의 토큰값들을 읽어와서 현재 로그인 상태인지 아닌지를 판단해야 한다고 생각했습니다. 제가 이전에는 토큰을 상태값으로 관리 안하고 필요할때 직접 로컬스토리지에 저장하고 삭제하는 방식이었거든요. 지금처럼 하나의 전역 상태로 관리하면 로컬스토리지 접근이 아예 불필요하네요 수정하겠습니다!

Reply:
@yrk721 
유래님 근데 이부분 코드를 수정하려다보니까..
그냥 initializeAuth를 선언하고 호출하는 코드 자체가 필요 없다고 느껴집니다..!
어차피 모든건 zustand와 연동될테니..
.
.
라고 생각했는데 그럼에도 불구하고 현재 상태를 한번 더 체크를 하는 이유는 로컬 스토리지에 토큰값은 없는데 isLoggedIn만 true인 인위적인 상태가 발생할 수 있다고 생각하고 남겨두긴 하겠습니다! (그럴 일은 거의,아예 없겠지만)

Reply:
✅ be677dccef09776d5a85f0a6c5075f3856bd58ec